lighthouse-core/audits/seo/meta-description.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+
+class Description extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Content Best Practices',
+      name: 'meta-description',
+      description: 'Document has a meta description',
+      failureDescription: 'Document does not have a meta description',
+      helpText: 'Meta descriptions may be included in search results to concisely summarize ' +
+          'page content. Read more in the ' +
+          '[Search Console Help page](https://support.google.com/webmasters/answer/35624?hl=en#1).',
+      requiredArtifacts: ['Description']
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    return {
+      rawValue: (artifacts.Description !== null && artifacts.Description.length > 0)
```

Comment:
1) Should we also `.trim()` it?
2) Should we include additional text explaining that either description was not found or it was empty?
3) BTW I've spotted that returning a numeric value here causes test to fail but green tick to be shown next to the test:
<img width="828" alt="screen shot 2017-09-01 at 21 21 45" src="https://user-images.githubusercontent.com/985504/29986946-0c114c26-8f66-11e7-8279-d39c7f0e5dcf.png">
Is this a bug?

Reply:
1. trim sgtm
2. debugString for those cases would be nice
3. yes this seems like a bug :) there's some magic that happens if the rawValue is a number to auto-populate the score but even if that's producing something weird the UI should still be consistent.

Reply:
Thanks for help! Regarding no.3 - should I create an issue for it?

Reply:
Inconsistency does look like a bug. In the meantime, should be able to set `scoringMode: Audit.SCORING_MODES.NUMERIC` in the meta information (otherwise everything other than 100 is interpreted as failing)

Reply:
@brendankenny thanks! For this audit rawValue is a boolean so we are fine, but I'll keep your advice in mind for the next ones.

Reply:
If the meta description exists but is invalid, I'd also recommend including the markup in the debugString. For example:

```
Description text is empty: `<meta name="description" content="     ">`
```

It'll be helpful for developers to know where to look to fix the issue.

Reply:
@rviscomi I think that's a good idea, but I wonder if it won't be confusing to show `<meta name="description" content="     ">` while the actual tag on page may look totally different (e.g. `<META YADA="yada" CONTENT="     " name="description" />`)?